commit 2f149aad4cc429f69e6406202b765588c4c49a19
Author: Bernhard Rosenkraenzer <Bernhard.Rosenkranzer@linaro.org>
Date:   Thu Mar 7 10:13:14 2013 +0100

    tf_comm: Make inline assembly compatible with clang

    clang currently doesn't know it needs to assign an even numbered
    register when hitting
        ldrexd %0, [%1]
    it does understand
        ldrexd %0, %H0, [%1]

    Signed-off-by: Bernhard Rosenkraenzer <Bernhard.Rosenkranzer@linaro.org>

diff --git a/security/tf_driver/tf_comm.h b/security/tf_driver/tf_comm.h
index 8921dc1..518acb2 100644
--- a/security/tf_driver/tf_comm.h
+++ b/security/tf_driver/tf_comm.h
@@ -81,8 +81,8 @@ static inline void tf_write_reg64(void *comm_buffer, u64 value)
 	u64 tmp;

 	__asm__ __volatile__("@ tf_write_reg64\n"
-		"1:	ldrexd %0, [%2]\n"
-		"	strexd %0, %1, [%2]\n"
+		"1:	ldrexd %0, %H0, [%2]\n"
+		"	strexd %0, %1, %H1, [%2]\n"
 		"	teq    %0, #0\n"
 		"	bne    1b"
 		: "=&r" (tmp)
